plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -270,80 +251,34 @@ public int getAdminPort() {
   /**
    * Starts the development server.
    *
-   * @param runnables the path to directories that contain configuration files such as
-   *        appengine-web.xml
    * @param console the stream (Eclipse console) to send development server process output to
    * @param arguments JVM arguments to pass to the dev server
    */
-  void startDevServer(List<File> runnables, MessageConsoleStream console, List<String> vmArguments)
+  void startDevServer(DefaultRunConfiguration devServerRunConfiguration,
+      MessageConsoleStream console)
       throws CoreException {
     
-    checkAndSetPorts();  // Must be called before setting the STARTING state.
-    setServerState(IServer.STATE_STARTING);
+    // Check ports before setting the STARTING state.
+    Predicate<Integer> portProber = new Predicate<Integer>() {
 
-    // Create dev app server instance
-    initializeDevServer(console);
+      @Override
+      public boolean apply(Integer port) {
+        return port == null || SocketUtil.isPortInUse(port);
+      }
+    };
 
-    // Create run configuration
-    DefaultRunConfiguration devServerRunConfiguration = new DefaultRunConfiguration();
-    devServerRunConfiguration.setAutomaticRestart(false);
-    devServerRunConfiguration.setAppYamls(runnables);
-    devServerRunConfiguration.setHost(getServer().getHost());
-    devServerRunConfiguration.setPort(serverPort);
-    devServerRunConfiguration.setAdminPort(adminPort);
-    devServerRunConfiguration.setJvmFlags(vmArguments);
+    serverPort = checkPort(devServerRunConfiguration.getPort(), DEFAULT_SERVER_PORT, portProber);
+    adminPort = checkPort(devServerRunConfiguration.getAdminPort(), DEFAULT_ADMIN_PORT, portProber);
+    checkPort(devServerRunConfiguration.getApiPort(), DEFAULT_API_PORT, portProber);
```

Comment:
`checkPort` just checks if the port or its default is in use. The default values here are the actual defaults used by _devappserver_.  The code below that set the admin port was a mistake.  

I included checking of the apiPort just in case as the `RunConfiguration` object is passed in.